spec/posts/index_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :system do
+  before(:each) do
+    @user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
+    @post = Post.create(title: 'My post', text: 'This is my first post', author: @user)
+    @post2 = Post.create(title: 'My post 2.0', text: 'This is my second post', author: @user)
+    @comment = Comment.create(text: 'This is my first comment', author: @user, post: @post)
+    Like.create(author: @user, post: @post)
+  end
+
+  describe 'Post Index Page' do
+    before(:each) do
+      visit user_posts_path(@user.id)
+    end
+
+    it 'displays the user image' do
+      expect(page).to have_css("img[src*='https://unsplash.com/photos/F_-0BxGuVvo']")
+    end
+
+    it 'displays the user name' do
+      expect(page).to have_content('Tom')
+    end
+
+    it 'displays the correct number of posts' do
+      expect(page).to have_content('Posts: 2')
+    end
+
+    it 'displays the post title' do
+      expect(page).to have_content('My post')
+    end
+
+    it 'displays the post text' do
+      expect(page).to have_content('This is my first post')
+    end
```

Comment:
- [ ] You missed adding the below test for the `posts#index` page. Please make sure to add the test and it should pass without any errors.
  - `Tests that asserts you can see a section for pagination if there are more posts than fit on the view.`

Reply:
Thank you for the helpful suggestions @thecodechaser ! I completely missed that last test case and I was unaware fixing the n+1 problem should be done in the controller. :smile: